apps/src/gamelab/GameLab.js

```diff
@@ -0,0 +1,574 @@
+  Studio.executeQueue('when-up');
+  Studio.executeQueue('when-right');
+  Studio.executeQueue('when-down');
+*/
+
+  if (this.JSInterpreter) {
+    this.JSInterpreter.executeInterpreter(this.tickCount === 1);
+  }
+
+/*
+  var currentTime = new Date().getTime();
+
+  if (!Studio.succeededTime && checkFinished()) {
+    Studio.succeededTime = currentTime;
+  }
+
+  if (!animationOnlyFrame) {
+    Studio.executeQueue('whenTouchGoal');
+  }
+
+  if (Studio.succeededTime &&
+      !spritesNeedMoreAnimationFrames &&
+      (!level.delayCompletion || currentTime > Studio.succeededTime + level.delayCompletion)) {
+    Studio.onPuzzleComplete();
+  }
+
+  // We want to make sure any queued event code related to all goals being visited is executed
+  // before we evaluate conditions related to this event.  For example, if score is incremented
+  // as a result of all goals being visited, recording allGoalsVisited here allows the score
+  // to be incremented before we check for a completion condition that looks for both all
+  // goals visited, and the incremented score, on the next tick.
+  if (Studio.allGoalsVisited()) {
+    Studio.trackedBehavior.allGoalsVisited = true;
+  }
+
+  // And we don't want a timeout to be used in evaluating conditions before the all goals visited
+  // events are processed (as described above), so also record that here.  This is particularly
+  // relevant to levels which "time out" immediately when all when_run code is complete.
+  if (Studio.timedOut()) {
+    Studio.trackedBehavior.timedOut = true;
+  }
+*/
+};
+
+GameLab.prototype.handleExecutionError = function (err, lineNumber) {
+/*
+  if (!lineNumber && err instanceof SyntaxError) {
+    // syntax errors came before execution (during parsing), so we need
+    // to determine the proper line number by looking at the exception
+    lineNumber = err.loc.line;
+    // Now select this location in the editor, since we know we didn't hit
+    // this while executing (in which case, it would already have been selected)
+
+    codegen.selectEditorRowColError(studioApp.editor, lineNumber - 1, err.loc.column);
+  }
+  if (Studio.JSInterpreter) {
+    // Select code that just executed:
+    Studio.JSInterpreter.selectCurrentCode("ace_error");
+    // Grab line number if we don't have one already:
+    if (!lineNumber) {
+      lineNumber = 1 + Studio.JSInterpreter.getNearestUserCodeLine();
+    }
+  }
+  outputError(String(err), ErrorLevel.ERROR, lineNumber);
+  Studio.executionError = { err: err, lineNumber: lineNumber };
+
+  // Call onPuzzleComplete() if syntax error or any time we're not on a freeplay level:
+  if (err instanceof SyntaxError) {
+    // Mark preExecutionFailure and testResults immediately so that an error
+    // message always appears, even on freeplay:
+    Studio.preExecutionFailure = true;
+    Studio.testResults = TestResults.SYNTAX_ERROR_FAIL;
+    Studio.onPuzzleComplete();
+  } else if (!level.freePlay) {
+    Studio.onPuzzleComplete();
+  }
```

Comment:
I was trying to leave the pieces in that I thought I would probably need to add back in some form. Since it took a while to prune out the parts that were app-specific, I was leaving them in until I decided.